src/main/java/me/imoltres/bbu/game/threads/GameThread.kt

```diff
@@ -57,7 +73,57 @@ class GameThread(val game: Game) : Thread() {
                             shrinkBorder(getBorderShrinkTime(border))
                             shrinking = !shrinking
                         }
+                        //check and see if any teams beacons are out of the border
+                        for (team in game.getTeams(true)) {
+                            val world = Bukkit.getWorld("world")
+                            val beaconLoc = team.beacon!!.toWorldPosition("world").toBukkitLocation()
+                            if (team.beacon != null) {
+                                if (beaconLoc.distance(world!!.worldBorder.center) > world.worldBorder.size / 2) {
+                                    team.beacon!!.toWorldPosition(BBU.getInstance().game.overworld.name).block.type =
+                                        Material.AIR
+                                    team.beacon = null
+                                    Bukkit.broadcast(
+                                        CC.translate("${team.getRawDisplayName()}&c's beacon has been destroyed because it was out of the border.")
+                                    )
+                                }
+                            }
+                        }
+                    }
+
+                    GameState.DEATH_MATCH -> {
+                        if (!deathmatch) {
+                            PlayerUtils.broadcastTitle(
+                                "&cDeath Match has started!",
+                                "&7"
+                            )
+
+                            for (team in game.getAliveTeams()) {
+                                team.beacon!!.toWorldPosition(BBU.getInstance().game.overworld.name).block.type =
+                                    Material.AIR
+                                team.beacon = null
+
+                                for (player in team.players) {
+                                    // *IMPORTANT* This is where the death match world is set
+                                    // May cause entity cramming if too many players are alive, but eh
+                                    // TODO make this configurable
+                                    player.player!!.teleport(Location(Bukkit.getWorld("bbuDeathMatchWorld"), 0.0, 100.0, 0.0))
```

Comment:
This will cause IMMENSE lag, thinking back to 32 player BBU, the servers crashed immediately because all players getting teleported into a condensed spot + it's not extremely necessary for the death match world.